<launch>
  <!-- Launch Turtle Sim -->
  <node name="roaming_turtle" pkg="turtlesim" type="turtlesim_node"/>

  <!-- Load in parameters from yaml file -->
  <rosparam command="load" file="$(find tsim)/config/turtle_rect_params.yaml"/>
  <rosparam command="load" file="$(find nuturtle_description)/config/diff_params.yaml"/>

  <!-- Launch the node to control the turtle -->
  <node name="turtle_way" pkg="tsim" type="turtle_way" output="screen"/>

  <!-- Defualt: launches a plot of the positional error -->
  <arg name="plot_gui" default="True" doc="A plot of the positional error will be displayed upon launch. Set to False to disable."/>
  <group if="$(arg plot_gui)">
    <node name="error_plot" pkg="rqt_plot" type="rqt_plot" args="/pose_error" />
  </group>

</launch>
